File: src/main/java/javabot/operations/JSROperation.java
package javabot.operations;

import com.antwerkz.sofia.Sofia;
import javabot.Message;
import javabot.operations.locator.JCPJSRLocator;
import org.pircbotx.Channel;

import javax.inject.Inject;

public class JSROperation extends BotOperation {
    @Inject
    JCPJSRLocator locator;

    @Override
    public boolean handleMessage(final Message event) {
        final String message = event.getValue().toLowerCase();
        final Channel channel = event.getChannel();
        if ("jsr".equals(message)) {
            getBot().postMessageToChannel(event, Sofia.jsrMissing());
            return true;
        } else {
            if (message.startsWith("jsr ")) {
                final String jsrString = message.substring("jsr ".length());

                try {
                    final int jsr = Integer.parseInt(jsrString);
                    String response = locator.findInformation(jsr);
                    if (response != null && !response.isEmpty()) {
                        getBot().postMessageToChannel(event, response);
                    } else {
                        getBot().postMessageToChannel(event, Sofia.jsrUnknown(jsrString));
                    }
                } catch (NumberFormatException nfe) {
                    getBot().postMessageToChannel(event, Sofia.jsrInvalid(jsrString));
                }
                return true;
            }
        }
        return false;
    }
}

File: src/main/kotlin/javabot/operations/JSROperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import java.util.Locale
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.service.JCPJSRLocator
import javax.inject.Inject

class JSROperation @Inject constructor(bot: Javabot, adminDao: AdminDao, var locator: JCPJSRLocator) : BotOperation(bot, adminDao) {
    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value.lowercase(Locale.getDefault())
        if ("jsr" == message.trim()) {
            responses.add(Message(event, Sofia.jsrMissing()))
        } else {
            if (message.startsWith("jsr ")) {
                val jsrString = message.substring("jsr ".length)

                try {
                    val jsr = Integer.parseInt(jsrString)
                    val response = locator.findInformation(jsr)
                    responses.add(Message(event, if (!response.isEmpty()) response else Sofia.jsrUnknown(jsrString)))
                } catch (nfe: NumberFormatException) {
                    responses.add(Message(event, Sofia.jsrInvalid(jsrString)))
                }
            }
        }

        return responses
    }
}

